docs/core/compatibility/interop/8.0/linux-netcoredeps.md

```diff
@@ -0,0 +1,45 @@
+---
+title: "Breaking change - Linux native library resolution no longer uses `netcoredeps`"
+description: "Learn about the breaking change in .NET 8 where Linux applications no longer search the `netcoredeps` subdirectory for native libraries."
+ms.date: 4/10/2025
+ai-usage: ai-assisted
+ms.custom: https://github.com/dotnet/docs/issues/45777
+---
+
+# Linux native library resolution no longer uses `netcoredeps`
+
+Starting in .NET 8, Linux applications no longer search the `netcoredeps` subdirectory for native libraries.
+
+## Version introduced
+
+.NET 8
+
+## Previous behavior
+
+In earlier versions of .NET, Linux applications searched for native libraries in a `netcoredeps` subdirectory located next to the application executable. This behavior applied to all native library loads, including user-defined platform invokes (p/invokes).
+
+## New behavior
+
+In .NET 8 and later, Linux applications no longer search the `netcoredeps` subdirectory for native libraries. Native library resolution now follows standard mechanisms without relying on this subdirectory.
+
+## Type of breaking change
+
+This is a [source incompatible](../../categories.md#source-compatibility) change.
+
+## Reason for change
+
+The `netcoredeps` behavior was originally introduced to handle complex dependencies on third-party libraries in earlier .NET versions. Modern .NET versions no longer require this behavior due to improved dependency handling. Additionally, the mechanism isn't aligned with recommended practices for ELF platforms.
+
+For more information, see [GitHub issue #114393](https://github.com/dotnet/runtime/issues/114393).
+
+## Recommended action
+
+If your application relied on the `netcoredeps` subdirectory for p/invokes or custom native library resolution, use the following alternatives:
+
+- Implement a custom resolution mechanism using <xref:System.Runtime.InteropServices.DllImportResolver> or the <xref:System.Runtime.Loader.AssemblyLoadContext.ResolvingUnmanagedDll> event.
```

Comment:
```suggestion
- Implement a custom resolution mechanism using the <xref:System.Runtime.Loader.AssemblyLoadContext.ResolvingUnmanagedDll> event or <xref:System.Runtime.InteropServices.NativeLibrary.SetDllImportResolver>.
```
- ResolvingUnmanagedDll is more likely to be solution that people are looking for, so I would mention it first
- DllImportResolver is just a delegate type that is not useful on its own. SetDllImportResolver docs explain what to do actually.